frontend/utils/image.js

```diff
@@ -1,7 +1,19 @@
 import { blake32Hash } from '~/shared/functions.js'
 import { handleFetchResult } from '~/frontend/controller/utils/misc.js'
 
-const imageUpload = async (imageFile) => {
+// Copied from https://stackoverflow.com/a/27980815/4737729
+export function b64toBlob (dataURI) {
+  var byteString = atob(dataURI.split(',')[1])
+  var ab = new ArrayBuffer(byteString.length)
+  var ia = new Uint8Array(ab)
+
+  for (var i = 0; i < byteString.length; i++) {
+    ia[i] = byteString.charCodeAt(i)
+  }
+  return new Blob([ab], { type: 'image/jpeg' })
```

Comment:
Please copy this part of from the StackOverflow answer into a comment between lines 4 and 5:

```
// Accepts only JPEGs and assumes dataURI begins with: "data:image/jpeg;base64,"
```

This will not work if, for example, a PNG is used.

Also, since `b64toBlob` sounds like it will accept any Base64 text and convert it to a Blob, this function should be renamed since that is not what it does. This function takes a JPEG data URI and converts it to a blob, so a better name would be something like: `jpegDataURItoBlob`.

Alternatively, if you parse the `dataURI` for the image type and use it here, you can give the function a more generic name, like `imageDataURItoBlob`

Reply:
Gotcha. Renamed to `imageDataURItoBlob` and added a TODO for later to support other formats :)